web3/utils/abi.py

```diff
@@ -472,12 +661,17 @@ def check_if_arguments_can_be_encoded(
     )
 
 
+@deprecated_for("get_abi_element")
```

Comment:
We should add a `with pytest.warns()` test for this and assert it doesn't warn when we use `get_abi_element()`. We've had a few issues where the warning wasn't quite working as expected recently.